### PR TITLE
Conversion of mobilevit to Keras 3 with tf.keras backwards compatibility

### PR DESCRIPTION
**Pull request by pksX01**
Conversion of [mobilevit.py](https://keras.io/examples/vision/mobilevit/) to Keras 3 with tensorflow backend as per stage 1 of **Issue: Keras.io examples conversion gameplan** #18468.

Provided solutions of the all problems raised by me in Issue #18613 .

### CODE REVIEW

File: examples/keras_io/tensorflow/vision/mobilevit.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+[original paper](https://arxiv.org/abs/2110.02178)):
+
+
+![](https://i.imgur.com/mANnhI7.png)
+"""
+
+
+def conv_block(x, filters=16, kernel_size=3, strides=2):
+    conv_layer = layers.Conv2D(
+        filters, kernel_size, strides=strides, activation=tf.nn.swish, padding="same"
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/mobilevit.py:75]: Use `"swish"`

File: examples/keras_io/tensorflow/vision/mobilevit.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    m = layers.DepthwiseConv2D(
+        3, strides=strides, padding="same" if strides == 1 else "valid", use_bias=False
+    )(m)
+    m = layers.BatchNormalization()(m)
+    m = keras.activations.swish(m)
+
+    m = layers.Conv2D(output_channels, 1, padding="same", use_bias=False)(m)
+    m = layers.BatchNormalization()(m)
+
+    if tf.math.equal(x.shape[-1], output_channels) and strides == 1:
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/mobilevit.py:99]: Use `if x.shape[-1] == output_channels`

File: examples/keras_io/tensorflow/vision/mobilevit.py [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+
+"""
+## Imports
+"""
+import os
+os.environ["KERAS_BACKEND"] = "tensorflow"
+
+import tensorflow as tf
+
+# For versions <TF2.13 change the above import to:
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/mobilevit.py:38]: Remove these 2 lines

File: examples/keras_io/tensorflow/vision/mobilevit.py [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+in global representations. With convolutions, we can capture spatial relationships that
+model locality.
+
+Besides combining the properties of Transformers and convolutions, the authors introduce
+MobileViT as a general-purpose mobile-friendly backbone for different image recognition
+tasks. Their findings suggest that, performance-wise, MobileViT is better than other
+models with the same or higher complexity ([MobileNetV3](https://arxiv.org/abs/1905.02244),
+for example), while being efficient on mobile devices.
+
+Note: This example should be run with Tensorflow 2.13 and higher.
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/mobilevit.py:27]: Remove outdated line

File: examples/keras_io/tensorflow/vision/mobilevit.py [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+]
+tflite_model = converter.convert()
+open("mobilevit_xxs.tflite", "wb").write(tflite_model)
+
+"""
+To learn more about different quantization recipes available in TFLite and running
+inference with TFLite models, check out
+[this official resource](https://www.tensorflow.org/lite/performance/post_training_quantization).
+
+You can use the trained model hosted on [Hugging Face Hub](https://huggingface.co/keras-io/mobile-vit-xxs)
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/mobilevit.py:378]: Remove these 2 outdated lines

File: examples/keras_io/tensorflow/vision/mobilevit.py [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+# Convert to TFLite. This form of quantization is called
+# post-training dynamic-range quantization in TFLite.
+converter = tf.lite.TFLiteConverter.from_saved_model("mobilevit_xxs")
+converter.optimizations = [tf.lite.Optimize.DEFAULT]
+converter.target_spec.supported_ops = [
+    tf.lite.OpsSet.TFLITE_BUILTINS,  # Enable TensorFlow Lite ops.
+    tf.lite.OpsSet.SELECT_TF_OPS,  # Enable TensorFlow ops.
+]
+tflite_model = converter.convert()
+open("mobilevit_xxs.tflite", "wb").write(tflite_model)
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/mobilevit.py:371]: Did you check that the TFLite model works?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/keras_io/tensorflow/vision/mobilevit.py (hunk 6 above)
Comment (pksX01) [reply examples/keras_io/tensorflow/vision/mobilevit.py:371]: I didn't perform the inference using TFLite model, but I verified that this line is saving TFLite model.

Comment (fchollet) [reply examples/keras_io/tensorflow/vision/mobilevit.py:371]: Can you try using the TFLite interpreter to check that it works as expected? https://www.tensorflow.org/lite/guide/inference#load_and_run_a_model_in_python

**Comment by pksX01** [comment]
@fchollet Completed all the changes asked by you. Also, I checked converted TFLite model is working as expected.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Comment by fchollet** [comment]
I'm not able to render the example as a notebook, the kernel dies at the line that writes the serialized model (even with 60GB of RAM). Are you able to run `python autogen.py add_example` on the file?

**Comment by pksX01** [comment]
> I'm not able to render the example as a notebook, the kernel dies at the line that writes the serialized model (even with 60GB of RAM). Are you able to run `python autogen.py add_example` on the file?

I couldn't find `autogen.py` in this repo. But I tried to run using `keras-io` repo which is causing some different error even on the original script.

**Comment by pksX01** [comment]
@fchollet Since I have made changes to work with tensorflow backend and it's merged  to be available in `examples/keras_io/tensorflow` , I wanted to do changes to make it backend agnostic. Can I leave tf.image functions like (resize, random_crop, random_flip_left_right etc.) as it is? Or I need to use Keras processing layers for them to make them backend agnostic? Please suggest.

**Pull request merged**